challenge/tlsalpn01/tls_alpn_challenge_test.go

```diff
@@ -116,3 +119,75 @@ func TestChallengeInvalidPort(t *testing.T) {
 	assert.Contains(t, err.Error(), "invalid port")
 	assert.Contains(t, err.Error(), "123456")
 }
+
+func TestChallengeIPaddress(t *testing.T) {
+	_, apiURL := tester.SetupFakeAPI(t)
+
+	domain := "127.0.0.1"
+	port := "23457"
+
+	mockValidate := func(_ *api.Core, _ string, chlng acme.Challenge) error {
+		conn, err := tls.Dial("tcp", net.JoinHostPort(domain, port), &tls.Config{
+			InsecureSkipVerify: true,
+		})
+		require.NoError(t, err, "Expected to connect to challenge server without an error")
+
+		// Expect the server to only return one certificate
+		connState := conn.ConnectionState()
+		assert.Len(t, connState.PeerCertificates, 1, "Expected the challenge server to return exactly one certificate")
+
+		remoteCert := connState.PeerCertificates[0]
```

Comment:
[RFC8738, section 6](https://datatracker.ietf.org/doc/html/rfc8738#section-6) requires the SNI extension to contain the `.in-addr.arpa` or `.ip6.arpa` reverse mapping of the IP address.

Can you add a test for this? [`(miekg/dns).ReverseAddr`](https://pkg.go.dev/github.com/miekg/dns#ReverseAddr) will be of help here.

Reply:
don't think we need that here, SNI extension is something ACME server's VA should sent, and out TLS-ALPN solver don't care about incomming traffics SNI and just give same certificate

Reply:
Hm. My thinking is: if the server MUST send the IP address in the SNI extension, the client should validate it is correct. The `http-01` challenge solver does something similar with the `Host`/`Forwarded-*` headers (this is mostly to prevent DNS rebind attacks though).

@ldez: Do you have an opinion on this?